src/binder/clrprivbinderassemblyloadcontext.cpp

```diff
@@ -251,9 +271,51 @@ HRESULT CLRPrivBinderAssemblyLoadContext::SetupContext(DWORD      dwAppDomainId,
     return hr;
 }
 
+/* static */
+void CLRPrivBinderAssemblyLoadContext::DestroyContext(CLRPrivBinderAssemblyLoadContext *pBindContext, INT_PTR ptrManagedStrongAssemblyLoadContext)
+{
+    CONTRACTL
+    {
+        GC_NOTRIGGER;
+        THROWS;
+        MODE_PREEMPTIVE;
+        SO_TOLERANT;
+    }
+    CONTRACTL_END;
+
+    GCX_COOP();
+    // Replace the weak handle with a strong handle
+    // in order to be able to callback Unloading safely
+    OBJECTHANDLE handle = reinterpret_cast<OBJECTHANDLE>(pBindContext->m_ptrManagedAssemblyLoadContext);
+    OBJECTHANDLE strongHandle = reinterpret_cast<OBJECTHANDLE>(ptrManagedStrongAssemblyLoadContext);
+    HndDestroyHandle(HndGetHandleTable(handle), HNDTYPE_WEAK_SHORT, handle);
+    pBindContext->m_ptrManagedAssemblyLoadContext = reinterpret_cast<INT_PTR>(strongHandle);
+
+    // We cannot delete the binder here as it is used indirectly when comparing assemblies with the same binder
+    // It will be deleted by the LoaderAllocator once
+#if !defined(DACCESS_COMPILE) && !defined(CROSSGEN_COMPILE)
+    ((AssemblyLoaderAllocator*)pBindContext->m_pAssemblyLoaderAllocator)->SetBinderToRelease(pBindContext);
```

Comment:
During initialization of LoadContext, aren't the LoadContext and LoadAllocator updated to keep mutual references (since that would alleviate the need to SetBinderToRelease as LoadAllocator would already know which LoadContext it needs to release)?